### DISCUSSION TITLE
Razor Page SelectList not populating when using ViewModel

### DISCUSSION THREAD

**Discussion by kmichelle7601** [dotnet/AspNetCore.Docs#26228] [Q&A]
Hi, 
I feel like I'm doing exactly what the tutorial is saying in https://docs.microsoft.com/en-us/aspnet/core/data/ef-rp/update-related-data?view=aspnetcore-6.0 but my query to populate my select list is not returning any values.  I am wondering if it's because I don't have foreign key relations on the database I inherited.  

I see this in the Contuso sample but not sure if this is because the database is being created within the sample tutorial.    Do I need this in my project too? Is this even related to my possible issue?

 protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Course>().ToTable(nameof(Course))
                .HasMany(c => c.Instructors)
                .WithMany(i => i.Courses);
            modelBuilder.Entity<Student>().ToTable(nameof(Student));
            modelBuilder.Entity<Instructor>().ToTable(nameof(Instructor));
        }

**Comment by dhanur350** [comment]
Exactly, where is your foreign key in this code, without that you cannot get  values return in your select list

**Reply by kmichelle7601** [reply]
The database that EF scaffolded doesn't have foreign keys.  Can I add the foreign keys in the model after the scaffold?

**Reply by dhanur350** [reply]
Yes you can by using core command, dotnet ef migrations add key
